tests/testthat/test-select_helpers.R

```diff
@@ -291,6 +305,23 @@ test_that("select_helpers: .formula_list_to_named_list ", {
     .formula_list_to_named_list(list(~ "Age", `age:trt` = "interact"), var_info = tidy_mod),
     list(age = "Age", trt = "Age", grade = "Age", `age:trt` = "interact")
   )
+
+  expect_error(
+    .formula_list_to_named_list(list(age ~ "Age"), var_info = tidy_mod, type_check = is.logical,
+                                arg_name = "label")
+  )
+  expect_error(
+    .formula_list_to_named_list(list(age ~ "Age"), var_info = tidy_mod,
+                                type_check = is.logical, arg_name = "label",
+                                type_check_msg = "Age msg error")
+  )
+  expect_error(
+    .formula_list_to_named_list("Age", var_info = tidy_mod,
+                                type_check = rlang::is_string,
+                                arg_name = "label",
+                                type_check_msg = "Age msg error")
+  )
```

Comment:
Ibid

Reply:
![image](https://user-images.githubusercontent.com/26774684/144714725-1bdb6538-0004-4bcc-b73b-ac383dd3d153.png)

This is the error message, but I was having trouble getting a regex match for the message. Do you know why this doesn't work?

```r
d>   expect_error(
+     .formula_list_to_named_list("Age", var_info = tidy_mod,
+                                 type_check = rlang::is_string,
+                                 arg_name = "label",
+                                 type_check_msg = "Age msg error"),
+     "*Did you mean `everything() ~ \"Age\"`\\?$"
+   )
Error: `.formula_list_to_named_list(...)` threw an error with unexpected message.
Expected match: "*Did you mean `everything() ~ \"Age\"`\\?$"
Actual message: "Error processing the `label` argument. Expecting a list or formula.\nReview syntax details at'https://www.danieldsjoberg.com/gtsummary/reference/syntax.html'\n\nDid you mean `everything() ~ \"Age\"`?"
```

Reply:
This is normal because it was not a formula.

Reply:
This is testing the error message as well. Here the logic is testing that the input is indeed a list (or a formula). But in this example the user passed the string `"Age"`. I added logic that if the users passes an incorrect type (ie not a list or a formula) AND the type of the passed object matched the `type_check=` predicate, the message asks the user 'Did you mean `everything() ~ "Age"`?'